### PR TITLE
Add some more examples of opaque types.

### PR DESCRIPTION
**Pull request by non**
I still wanted to write more, but here's what I have at this moment.

@jvican I'm going to try to push some more change to this branch but feel free to merge what you want when you want.

### CODE REVIEW

File: _sips/sips/2017-09-20-opaque-types.md [hunk 1]
```diff
@@ -376,6 +376,272 @@ The extension methods synthesized for implicit value classes are not static. We
...
+      Ordering.by(_ => ())
+  }
+}
+```
+
+### Probability interval
+
+```scala
+package object prob {
+  opaque type Probability = Double
```
Comment (jvican) [inline _sips/sips/2017-09-20-opaque-types.md]: This is a very nice example.

File: _sips/sips/2017-09-20-opaque-types.md [hunk 2]
```diff
@@ -376,6 +376,272 @@ The extension methods synthesized for implicit value classes are not static. We
...
+
+It would be possible to automatically provide `ClassTag` instances,
+using an `implicit val` in the case of opaque types wrapping concrete
+types (e.g.`opaque type X = Double`) and `implicit def` in cases such
+as above.
+
+### Fix-point type
+
+```scala
+package object fixed {
```
Comment (jvican) [inline _sips/sips/2017-09-20-opaque-types.md]: 👏

**Comment by jvican** [comment]
My updates: I had a read of http://cr.openjdk.java.net/~jrose/values/shady-values.html and I have nice observations to make on the differences between value types and opaque types. I'll follow up with a writeup of these!

**Comment by jvican** [comment]
We can wait and add updates in this PR before next SIP meeting (in November), or merge this right away and follow up in different PRs. What do you think it's best @non?

**Comment by non** [comment]
I just added a `Nullable` example that uses `AnyRef` as a bound.

**Pull request merged**